composer/spec/dependabot/composer/update_checker/version_resolver_spec.rb

```diff
@@ -274,21 +274,22 @@
       end
     end
 
-    # This test is extremely slow, as it needs to wait for Composer to time out.
-    # As a result we currently keep it commented out.
-    # context "with an unreachable private registry" do
-    #   let(:project_name) { "unreachable_private_registry" }
-    #   let(:dependency_name) { "dependabot/dummy-pkg-a" }
-    #   let(:dependency_version) { nil }
-    #   let(:string_req) { "*" }
-    #   let(:latest_allowable_version) { Gem::Version.new("6.0.0") }
-
-    #   it "raises a Dependabot::PrivateSourceTimedOut error" do
-    #     expect { resolver.latest_resolvable_version }.
-    #       to raise_error(Dependabot::PrivateSourceTimedOut) do |error|
-    #         expect(error.source).to eq("https://composer.dependabot.com")
-    #       end
-    #   end
-    # end
+    context "with an unreachable private registry" do
+      let(:project_name) { "unreachable_private_registry" }
+      let(:dependency_name) { "dependabot/dummy-pkg-a" }
+      let(:dependency_version) { nil }
+      let(:string_req) { "*" }
+      let(:latest_allowable_version) { Gem::Version.new("6.0.0") }
+
+      before { ENV["COMPOSER_PROCESS_TIMEOUT"] = "1" }
+      after { ENV.delete("COMPOSER_PROCESS_TIMEOUT") }
+
+      it "raises a Dependabot::PrivateSourceTimedOut error" do
+        expect { resolver.latest_resolvable_version }.
+          to raise_error(Dependabot::PrivateSourceTimedOut) do |error|
+            expect(error.source).to eq("https://composer.dependabot.com")
```

Comment:
Rather than hanging and eventually returning a timeout error, this returns:
```ruby
expected Dependabot::PrivateSourceTimedOut, got #<Dependabot::SharedHelpers::HelperSubprocessFailed: curl error 6 while downloading https://composer.dependabot.com/packages.json: Could not resolve host: composer.dependabot.com>
```

A few related changes since this test was originally written:
* `https://composer.dependabot.com` no longer has a DNS record now that Dependabot has been acquired by GitHub
* This was originally written against `composer` `v1`, and we're now using `v2` so may return the resolving failure differently


But in order to test this, we need to mock a routable endpoint that simply hangs... while I'm sure an easy way to do this exists in Ruby, I'm not sure how?

Reply:
There's some good suggestions here: https://stackoverflow.com/questions/100841/artificially-create-a-connection-timeout-error

Reply:
I'm going to handle ☝️ in a follow-up PR, as I expect once I fix the test we'll have to also update the code... so for now let's get the test marked pending so it's at least on our radar as something that needs fixing and not hiding commented out.

Reply:
https://github.com/dependabot/dependabot-core/pull/6776